# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: 85194e5b7a9e
# Full Hash: 85194e5b7a9e6ed9114ee46aef0b5efb819cca9b
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-04 03:56:33
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part9: Update DOMLocalization to match Localization constructor. r=nika,emilio
#   
#   Depends on D113691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113692
# ==============================================================================

diff -r 181487bdf6c0 -r 85194e5b7a9e dom/base/Document.cpp
--- a/dom/base/Document.cpp	Tue Aug 03 16:25:13 2021 +0000
+++ b/dom/base/Document.cpp	Tue Aug 03 16:25:13 2021 +0000
@@ -4173,7 +4173,7 @@
     mDocumentL10n = DocumentL10n::Create(this, isSync);
     MOZ_ASSERT(mDocumentL10n);
   }
-  mDocumentL10n->AddResourceId(href);
+  mDocumentL10n->AddResourceId(NS_ConvertUTF16toUTF8(href));
 
   if (mReadyState >= READYSTATE_INTERACTIVE) {
     mDocumentL10n->TriggerInitialTranslation();
@@ -4196,7 +4196,8 @@
   if (mDocumentL10n) {
     nsAutoString href;
     aLinkElement->GetAttr(kNameSpaceID_None, nsGkAtoms::href, href);
-    uint32_t remaining = mDocumentL10n->RemoveResourceId(href);
+    uint32_t remaining =
+        mDocumentL10n->RemoveResourceId(NS_ConvertUTF16toUTF8(href));
     if (remaining == 0) {
       if (mDocumentL10n->mBlockingLayout) {
         mDocumentL10n->mBlockingLayout = false;